src/app/services/result-of-test/result-of-test.service.ts

```diff
@@ -0,0 +1,21 @@
+import { Injectable } from '@angular/core';
+import { resultOfTestEnum } from '@models/test-types/test-type.model';
+import { select, Store } from '@ngrx/store';
+import { State } from '@store/.';
+import { resultOfTestSelector, updateResultOfTest } from '@store/test-records';
+import { Observable } from 'rxjs';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ResultOfTestService {
```

Comment:
This adheres to our current pattern and the decision to use this was made by someone else who is not in the project anymore, so it is fine. But just an observation here:

Both methods here are one-liners. The only thing this service achieves is to hide the `store`. In other words, abstraction for abstraction's sake. In object-oriented programming, abstraction is used to reduce complexity and increase efficiency. I'll use one example...

```typescript
  calculateResult(sequence: number[], exponent: number): number {
    const integral = this.mathsService.evaluateIntegralSequence(sequence); // contains several loops

    const fibonacci = this.mathsService.getNearestFibonacci(exponent); // works with a huge switch case

    return this.mathsService.recurseAndCombine(integral, fibonacci); // consists of several recursive methods
  }
```
Here the method is concerned with applying several steps and returning their result. The internals of those steps is long and complex. Without abstraction, this method could be 30 lines and its flow would not be apparent anymore.


---
```typescript
  getSum(a: number, b: number): number {
    return this.mathsService.addTwoNumbers(a, b);
  }

  class mathsService {
    addTwoNumbers(a: number, b: number): number {
      return a + b;
    }
  }
```
Whilst here, we are moving just one line of code from one place to another.

I think it would be a fruitful conversation to decide on this sometime in the future (before it's too late) @mcadenas-bjss  & @andrewf-bjss. And as a result, if the consensus is to still abstract with services, then I'm happy to comply - but at least we would have made that decision explicitly.